src/pages/PollsPage.js

```diff
@@ -1,24 +1,119 @@
-import React, { useEffect } from "react";
-import axiosInstance from "../utils/axiosInterceptor";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import PollItem from "../component/PollItem";
+import {
+  deleteSinglePoll,
+  getPollList,
+} from "../redux/reducer/pollListReducer";
+
+import ConfimationModal from "../component/ConfimationModal";
+import { ClipLoader, MoonLoader } from "react-spinners";
+import ChartModal from "../component/ChartModal";
 
 const PollsPage = () => {
-  const fetchUserData = async () => {
-    // Example GET request
-    axiosInstance
-      .post("/polling")
-      .then((response) => {
-        // Handle response data
-      })
-      .catch((error) => {
-        // Handle error
-      });
+  const [pageNumber, setPageNumber] = useState(1);
+  const [polls, setPolls] = useState([]);
+  const { pollList, loading } = useSelector((state) => state.pollList);
+  const [chartData, setChartData] = useState({
+    labels: [],
+    datasets: [
+      {
+        label: "",
+        data: [],
+        backgroundColor: "rgb(0, 137, 167)",
+        borderWidth: 1,
+      },
+    ],
+  });
+  const [showPollChart, setShowPollChart] = useState(false);
+  const [showDeletedModal, setShowDeletedModal] = useState(false);
+  const [selectedPoll, setSelectedPoll] = useState(null);
+  const dispatch = useDispatch();
+
+  const viewPollVoteChart = (pollData) => {
+    const optionList = pollData.optionList.map(
+      ({ optionTitle, voteCount }) => ({ optionTitle, voteCount })
+    );
+    const labels = optionList.map((items) => items.optionTitle);
+    setChartData({
```

Comment:
same for this
do you need this on PollsPage? It can be done on the chat page too @amitark7